### PR TITLE
calculator milestone four

### PR DESCRIPTION
**Pull request by codesalley**
## calculator milestone four

- added check cases to verify operators and numbers, and to make changes base on the given conditions.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,6 +18,11 @@
 .env.test.local
 .env.production.local
 
+# ignore congig files for heroku to work properly
+
+.eslintrc.json
```
Comment (bellom) [inline .gitignore]: You need to remove `.eslintrc.json` from this file, and also set up the recommended `.eslintrc.json`  check out the [microverse-linter-config](https://github.com/microverseinc/linters-config/tree/master/react-redux), according to the project requirement we need to set the linter's correctly with the standard Microverse config file without changing it.

File: src/components/App.js [hunk 2]
```diff
@@ -1,16 +1,23 @@
+/* eslint-disable no-unused-vars */
 import './App.css';
+import { useState } from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
+import isValidNumber from './logic/validNum';
```
Comment (bellom) [inline src/components/App.js:7]: I think you need to import `Calculate.js` here not in `ButtonPanel.js`
You might need to adjust the codebase to follow the instruction in the pathwright, we both know that all students follow these instructions.
image
[Screenshot from pathwright](https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106376018/)

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @juxsalley  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well adding `calculate.js`, `operate.js` and making use of `useState`, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**1.** It would be nice to always double-check your commit messages, you have some 
typo.

**2. Inline comment**
With the ESLint inline comment, you're are indirectly changing the linter's rules by disabling these rules, let's fix the codebase instead.

If you are using VSCode, you can delete the inline comment and hover over the error(configure eslint checks locally). And you will see your options. Click on the more info button that will take you to a GitHub repo where it is explained why you are getting this error. I find this feature very educational. Please, try to learn and implement the necessary rules so you don't have to keep this disabling line on top of each file. I believe you can do this! 💪😇

**3.** Kindly check out my comments below

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by arslanbisharat — APPROVED** [review]
# Approved :clap:

Hello Developer :wave:

Thank You for submitting this Pull Request. I am thrilled to inform you that your project is approved now. :clap:

You have done a great job. I like your project. 

**To highlight** 

- [ ] You have a descriptive Readme. Nice Job. 
- [ ] Your Pull Request is nice. Good Job!
- [ ] You have deployed your website. Nice!!!
- [ ] You are using linters and no error is present. I am happy to see it. 
- [ ] Your project has met the minimum technical requirements too. 



In short, I am very excited to see this project and proud of you. 

**What to do next?** :thinking:

1- Visit https://dashboard.microverse.org/projects and submit your approved project. 
2- Prepare yourself for the next project. 

I am wishing you very good luck with the next project. :smile:

Once again, Congratulations!:joy:

**Best,
Arslan Bisharat(PT-TSE)
Slack: @arslanbisharat**

**Pull request merged**